crates/payload/basic/src/builder_stack.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{
+    BuildArguments, BuildOutcome, MissingPayloadBehaviour, PayloadBuilder, PayloadBuilderError,
+    PayloadConfig,
+};
+
+/// A stack of payload builders that can be used in sequence.
+#[derive(Clone, Debug)]
+pub struct PayloadBuilderStack<B>(Vec<B>);
+
+impl<B> PayloadBuilderStack<B> {
+    /// Create a new payload builder stack.
+    pub fn new(builders: Vec<B>) -> Self {
+        Self(builders)
+    }
+
+    /// Push a builder onto the stack.
+    pub fn push(&mut self, builder: B) {
+        self.0.push(builder);
+    }
+}
+
+impl<B, Pool, Client> PayloadBuilder<Pool, Client> for PayloadBuilderStack<B>
+where
+    B: PayloadBuilder<Pool, Client>,
+{
+    type Attributes = B::Attributes;
+    type BuiltPayload = B::BuiltPayload;
```

Comment:
if we do `Stack<A,B>` then these could be `Either<A,b>`:

https://docs.rs/futures/latest/futures/future/enum.Either.html